ada/language/ast.py

```diff
@@ -9382,6 +9311,181 @@ class AggregateAssoc(BasicAssoc):
     expr = Property(Entity.r_expr)
     names = Property(Self.designators.map(lambda d: d))
 
+    xref_stop_resolution = Property(True)
+
+    base_aggregate = Property(
+        Entity.parent.parent.cast_or_raise(BaseAggregate)
+    )
+
+    @langkit_property()
+    def xref_equation():
+        agg = Var(Entity.base_aggregate)
+        return Cond(
+            agg.in_aspect('Global'), Entity.globals_assoc_equation,
+
+            agg.in_aspect('Depends'), Entity.depends_assoc_equation,
+
+            agg.parent.is_a(AspectClause, AspectAssoc, PragmaArgumentAssoc),
+            LogicTrue(),
+
+            Entity.general_xref_equation
+        )
+
+    @langkit_property(return_type=Equation, dynamic_vars=[env, origin])
+    def general_xref_equation():
+        """
+        Equation for an aggregate of a record or array.
+        """
+        mra = Var(Entity.base_aggregate.multidim_root_aggregate)
+
+        # If we're part of a multidim aggregate, then take the root aggregate's
+        # type. Else, this is a regular aggregate. In this case grab the type
+        # in type_val.
+        td = Var(If(
+            Not(mra.is_null),
+            mra.typ,
+            Entity.base_aggregate.type_val.cast(BaseTypeDecl),
+        ))
+
+        atd = Var(td.array_def)
+
+        return Cond(
```

Comment:
1. You have a `Cond` where an `If` would do
2. Seems to me like at this stage you can inline `general_xref_equation` inside `xref_equation`

Reply:
Done!